packages/template-retail-react-app/package.json

```diff
@@ -59,8 +59,8 @@
         "analyze-build": "cross-env MOBIFY_ANALYZE=true npm run build",
         "build-translations": "npm run extract-default-translations && npm run compile-translations",
         "compile-translations": "formatjs compile-folder --ast app/translations app/translations/compiled",
-        "compile-translations:pseudo": "formatjs compile --ast app/translations/$npm_package_l10n_defaultLocale.json --out-file app/translations/compiled/en-XB.json --pseudo-locale en-XB",
-        "extract-default-translations": "formatjs extract 'app/**/*.{js,jsx}' --out-file app/translations/$npm_package_l10n_defaultLocale.json --id-interpolation-pattern [sha512:contenthash:base64:6]",
+        "compile-translations:pseudo": "formatjs compile --ast app/translations/en-US.json --out-file app/translations/compiled/en-XB.json --pseudo-locale en-XB",
```

Comment:
This takes away the ability for users to customize their default locale. We don't need to use a field in the package.json, but we might want to consider having _some_ way for them to customize, like `$PWA_KIT_DEFAULT_LOCALE`. Not sure what the best solution would be, though, for providing a default/fallback value.

Reply:
@wjhsf That's a good point.. there's some context that I didn't mention in this PR: why did we decide to hardcode the default locale like that? Long story short.. it's not ideal, but it was the simplest solution at the moment. At one point, we were considering to have a new field in our config js file.

Here's another place where we hardcode the default locale: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/b0e503a0edee6d6ffa208c63a77af1d112987092/packages/template-retail-react-app/app/components/_app/index.jsx#L164-L168

There are 2 versions of default locales that we need consider:
- From demandware backend: there could be _multiple_ default locales - one for each site
- From the retail react app: it's the default locale for the hardcoded UI text - there's only 1 for the entire project.

Because of these differences, I'm currently not sure what the best way forward is. 

BUT for the scope of this PR, let's focus on making the localization scripts to be the same as what's in V1. Then we can have a separate PR / discussion to replace the hardcoding of this default locale.